privacyidea/models.py

```diff
@@ -204,25 +205,27 @@ def __init__(self, serial, tokentype=u"",
         self.otplen = otplen
         self.pin_seed = u""
         self.set_otpkey(otpkey)
-        self.resolver = None
-        self.resolver_type = None
-        self.user_id = None
-            
+
         # also create the user assignment
         if userid and resolver and realm:
-            # get type of resolver
-            res_type = Resolver.query.filter_by(name=resolver).first().rtype
-            self.resolver = resolver
-            self.resolver_type = res_type
-            self.user_id = userid
-            # We can not create the tokenrealm-connection, yet
+            # We can not create the tokenrealm-connection and owner-connection, yet
             # since we need to token_id.
             token_id = self.save()
             realm_id = Realm.query.filter_by(name=realm).first().id
             tr = TokenRealm(realm_id=realm_id, token_id=token_id)
             if tr:
                 db.session.add(tr)
+
+            # get type of resolver
+            res_type = Resolver.query.filter_by(name=resolver).first().rtype
+            to = TokenOwner(token_id=token_id, user_id=userid, resolver=resolver, resolver_type=res_type,
+                            realm_id=realm_id)
+            if to:
+                db.session.add(to)
```

Comment:
Could it ever be the case that ``to`` is false-y?

Reply:
I am not sure. It might be if some constraints of the tokenowner table are not met. (doublicate entry).
At a first glance it looks like you are problably right.
For certain reasons we use this way of code at several places - especially since ``db.session.add(None)`` would raise an exception.